addons/viewport/src/components/Panel.jsx

```diff
@@ -0,0 +1,120 @@
+/* global document */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { viewports, defaultViewport, resetViewport } from './viewportInfo';
+
+import { SelectViewport } from './SelectViewport';
+import { RotateViewport } from './RotateViewport';
+
+import * as styles from './styles';
+
+const storybookIframe = 'storybook-preview-iframe';
+const containerStyles = {
+  padding: 15,
+  width: '100%',
+  boxSizing: 'border-box',
+  fontFamily:
+    '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", Arial, sans-serif',
+};
+
+export class Panel extends Component {
+  static propTypes = {
+    channel: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+  };
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      viewport: defaultViewport,
+      isLandscape: false,
+    };
+
+    this.props.channel.on('addon:viewport:update', this.changeViewport);
+  }
+
+  componentDidMount() {
+    this.iframe = document.getElementById(storybookIframe);
+  }
+
+  componentWillUnmount() {
+    this.props.channel.removeListener('addon:viewport:update', this.changeViewport);
+  }
+
+  iframe = undefined;
+
+  changeViewport = viewport => {
+    const { viewport: previousViewport } = this.state;
+
+    if (previousViewport !== viewport) {
+      this.setState(
+        {
+          viewport,
+          isLandscape: false,
+        },
+        this.updateIframe
+      );
+    }
+  };
+
+  toggleLandscape = () => {
+    const { isLandscape } = this.state;
+
+    this.setState({ isLandscape: !isLandscape }, this.updateIframe);
+  };
+
+  updateIframe = () => {
```

Comment:
arrow-functions cannot have a this binding, why are they object methods then?

If they are utility functions that are unrelated to the instance, we should hoist them to the top of the module. Or possibly even be moved into their own module.

Reply:
The function obviously uses `this`. Making it arrow function is just one more way to make it suitable as a callback: https://facebook.github.io/react/docs/handling-events.html ("If calling `bind` annoys you" section)

Reply:
@ndelangen are you saying to instead have this be a normal class function than in the constructor do `this.updateIframe = this.updateIframe.bind(this);`? Happy to move to that pattern instead if you want, just want to make sure i do it correctly :)

fwiw, this is a fairly common pattern as @Hypnosphi pointed out, but I think it's more important to be consistent through the codebase.